interview-questions.md

```diff
@@ -8,44 +8,44 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Researched answer:
+Researched answer: Calling super is used to reference the superclass of the current method. It references the initialize method within the parent class, and acts ad if you were calling the the original method. An example would be there is a class CocaCola with a def initialize, and within that initialize there is a symbol of @flavor. Then the class DietCoke < CocaCola could call super to have access to @flavor, without remaking the symbol within class DietCoke.
 
 2. What is a gem?
 
-Your answer:
+Your answer: A gem is a tool used to install dependencies and packages containing pre-built data. 
 
-Researched answer:
+Researched answer: A gem is a library of data that has specific purposes. It can also contain files or dependencies required or related to that purpose. Gems are open source, which allows developers to have access to programs and data other developers have created.
 
 3. What is a relational database? Are there other kinds of databases?
 
-Your answer:
+Your answer: No idea
 
-Researched answer:
+Researched answer: A relational database is a type of database that structures data in the form of a table with rows and columns, such as PostgreSQL. There are many different types of databases. To name a few, there are centralized, cloud, graph, object-oriented databases, among many others.
 
 4. What are primary keys? Why are they important?
 
-Your answer:
+Your answer: Primary keys are specific ID values that are assigned to rows in relational databases. An example would be in Ruby on Rails, each row in a table is given a number value as its primary key. Each key can only be used once in each table, as it is a unique identifier.
 
-Researched answer:
+Researched answer: A primary key is a separate column in a relational database that is designated to be a unique identifier for each set of data in the table. For example, a table with data for states in the United States would have each state assigned a primary key. That primary key is then linked with its corresponding state and can be used to identify that state.
 
 5. What are the HTTP verbs? What is each verb's corresponding CRUD action?
 
-Your answer:
+Your answer: The HTTP verbs are Get, Put, Post, Patch, and Delete. Get goes with read, put goes with update, post and patch go with create, and delete goes with delete.
 
-Researched answer:
+Researched answer: The HTTP verbs are Get, Put, Post, Patch, and Delete. Get goes with read, put goes with update, post and patch go with create, and delete goes with delete.
 
```

Comment:
Recommend looking over this section in the syllabus. 
https://github.com/learn-academy-2022-hotel/Syllabus/blob/main/rails/restful-routes-crud.md
Create examples based on similar topic.